cmd/server/main.go

```diff
@@ -33,15 +36,31 @@ func main() {
 	if err != nil {
 		logger.Fatalln(err)
 	}
+
+	middleware := grpc_middleware.ChainUnaryServer(
+		grpc_validator.UnaryServerInterceptor(),                     // validation middleware
+		grpc_logrus.UnaryServerInterceptor(logrus.NewEntry(logger)), // logging middleware
+	)
+	// add authorization middleware if the pdp address is provided
+	if PDPAddr != "" {
+		authorizer := toolkit_auth.Authorizer{
+			PDPAddr,
+			toolkit_auth.NewBuilder(
+				toolkit_auth.WithJWT(nil),
+				toolkit_auth.WithRequest(),
+			),
+			toolkit_auth.NewHandler(),
+		}
+		middleware = grpc_middleware.ChainUnaryServer(
```

Comment:
also, it sucks to have to reproduce the entire middleware chain for just a single middleware. I imagine that if we have multiple flag-based middlewares (tracing, metrics, etc), this pattern would fall apart.

Maybe refactor to be a slice or something?

Reply:
Yeah, I completely agree. I'll update it to use a slice.